allegro.h

```diff
@@ -580,11 +580,14 @@ typedef class Serial_read_buffer : public Serial_buffer {
     double get_double() { assert(!(((intptr_t) ptr) & 7));
                           double d = *((double *) ptr); ptr += sizeof(double); 
                           return d; }
-    const char *get_string() { char *s = ptr; char *fence = buffer + len;
-                         assert(ptr < fence);
-                         while (*ptr++) assert(ptr < fence);
-                         get_pad();
-                         return s; }
+    const char *get_string() {
+        char *s = ptr;
+        [[maybe_unused]] char *fence = buffer + len;
```

Comment:
Hi! The `[[maybe_unused]]` attribute is from C++17, and since this is an older codebase, you'd probably want to use `(void)fence;` (cast to void) to retain C++98 compatibility.

Reply:
True. `(void)` casts are already used in a few other places in portsmf too so I'll just do that